tests/test_utils_torch.py

```diff
@@ -0,0 +1,188 @@
+            assert 'x' in right
+            assert 'y' in right
+            assert right['x'].shape == (8, 5)
+            assert right['y'].shape == (8, 5)
+
+    def test_split_emptydict(self):
+        # test a horrible edge case where d is an empty dict, and we need to
+        # return a BUNCH of empty dicts
+        t = torch.randn(32, 5)
+        d = {}
+        tup = (t, d)
+        items = PipelineHelper.split(tup, 8)
+        assert len(items) == 4
+        for item in items:
+            assert isinstance(item, tuple)
+            a, b = item
+            assert isinstance(a, torch.Tensor)
+            assert a.shape == (8, 5)
+            assert isinstance(b, dict)
+            assert b == {}
+
+    def test_split_emptydict(self):
+        # test an even worse edge case, where we have a dict of empty dicts
+        t = torch.randn(32, 5)
+        d = {'x': {}}
+        tup = (t, d)
+        items = PipelineHelper.split(tup, 8)
+        assert len(items) == 4
+        for item in items:
+            assert isinstance(item, tuple)
+            a, b = item
+            assert isinstance(a, torch.Tensor)
+            assert a.shape == (8, 5)
+            assert isinstance(b, dict)
+            assert b == {}
+
+    def test_join_tensor(self):
+        t = torch.randn(8, 5)
+        j = PipelineHelper.join([t, t, t, t])
+        assert isinstance(j, torch.Tensor)
+        assert j.shape == (32, 5)
+
+        j = PipelineHelper.join([t, t], dim=1)
+        assert isinstance(j, torch.Tensor)
+        assert j.shape == (8, 10)
+
+    def test_join_tuple(self):
+        tup = (torch.randn(8, 5), torch.randn(8, 2))
+        chunks = [tup, tup]
+        j = PipelineHelper.join(chunks)
+        assert isinstance(j, tuple)
+        assert len(j) == 2
+        a, b = j
+        assert a.shape == (16, 5)
+        assert b.shape == (16, 2)
+
+    def test_join_dict(self):
+        chunk = {'x': torch.randn(8, 5), 'y': torch.randn(8, 2)}
+        chunks = [chunk, chunk]
+        j = PipelineHelper.join(chunks)
+        assert isinstance(j, dict)
+        assert len(j) == 2
+        assert 'x' in j
+        assert 'y' in j
+        assert isinstance(j['x'], torch.Tensor)
+        assert isinstance(j['y'], torch.Tensor)
+        assert j['x'].shape == (16, 5)
+        assert j['y'].shape == (16, 2)
+
+    def test_join_complex(self):
+        d = {'x': torch.randn(8, 5), 'y': torch.randn(8, 2)}
+        t = torch.Tensor(8, 3)
+        tup = (t, d)
+        chunks = [tup, tup]
+        j = PipelineHelper.join(chunks)
+        assert isinstance(j, tuple)
+        assert len(j) == 2
+        left, right = j
+        assert isinstance(left, torch.Tensor)
+        assert left.shape == (16, 3)
+        assert isinstance(right, dict)
+        assert len(right) == 2
+        assert 'x' in right
+        assert 'y' in right
+        assert right['x'].shape == (16, 5)
+        assert right['y'].shape == (16, 2)
+
+    def chunk_layer_iterator(self):
```

Comment:
What's this for?

Reply:
Unwritten unit test. It WILL become the workitem iterator.